scan/spec/xcpretty_reporter_options_generator_spec.rb

```diff
@@ -0,0 +1,184 @@
+describe Scan do
+  describe Scan::XCPrettyReporterOptionsGenerator do
+    before(:all) do
+      options = { project: "./scan/examples/standard/app.xcodeproj" }
+      Scan.config = FastlaneCore::Configuration.create(Scan::Options.available_options, options)
+    end
+
+    describe "xcpretty reporter options generation" do
+      it "generates options for the junit tempfile report required by scan" do
+        generator = Scan::XCPrettyReporterOptionsGenerator.new(false, "html", "report.html", "test_output", false)
+        reporter_options = generator.generate_reporter_options
+        temp_junit_report = Scan.cache[:temp_junit_report]
+
+        expect(temp_junit_report).not_to be_nil
+        expect(reporter_options).to include("--report junit")
+        expect(reporter_options).to include("--output #{temp_junit_report}")
+      end
+
+      it "generates options for a custom junit report with default file name" do
```

Comment:
I haven't seen a test for the following case 
1. default output types are set (html,junit)
2. `custom_report_file_name` is set but not `output_files`

this should trigger the warning of the number of elements in `output_types` and `output_files` are not the same and use the default `report.#{type}` file name, if I understood the intent.